x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/cytoscape.tsx

```diff
@@ -76,7 +86,7 @@ export function Cytoscape({ children, elements, height, style, width }: Cytoscap
   const dataHandler = useCallback<cytoscape.EventHandler>(
     (event) => {
       if (cy && height > 0) {
-        cy.layout(getLayoutOptions(width, height)).run();
+        setTimeout(() => cy.layout(getLayoutOptions(width, height)).run(), 150);
```

Comment:
I think it'd be better to wrap `dataHandler` callback with `debounce` instead of using this `setTimeout`

Reply:
This is a short-term fix for the layout issue when there is only one 'row' of nodes. I'm not sure if `debounce` would be a better solution as it uses `setTimeout` under the hood anyway and there is no need to reduce the amount of function calls by delaying - we simply need to wait a bit before running the the layout.

Reply:
and how can we be sure 150ms suffices? 🤔  also from this condition, I don't see it covers only one corner case for 1 row